sdk/eventhub/event-hubs/src/eventPosition.ts

```diff
@@ -226,3 +82,54 @@ export function isEarliestEventPosition(eventPosition: EventPosition): boolean {
 
   return false;
 }
+
+/**
+ * Gets the `EventPosition` corresponding to the location of the the first event present in the partition.
+ * Use this position to begin receiving from the first event that was enqueued in the partition
+ * which has not expired due to the retention policy.
+ */
+export const earliestEventPosition: EventPosition = {
+  offset: -1
+}
+
+
+/**
+ * Gets the `EventPosition` corresponding to the end of the partition, where no more events are currently enqueued.
+ * Use this position to begin receiving from the next event to be enqueued in the partion after an ``EventHubConsumer``
+ * is created with this position.
+ * @returns EventPosition
+ */
+export const latestEventPosition: EventPosition = {
+  offset: "@latest"
+}
+
+/**
+ * @ignore
+ * @internal
+ */
+export function validateEventPosition(position: EventPosition) {
+  
+  const offsetPresent = position.offset != undefined;
+  const sequenceNumberPresent = position.sequenceNumber != undefined;
+  const enqueuedOnPresent = position.enqueuedOn != undefined;
+
+  if ((offsetPresent && sequenceNumberPresent) || (offsetPresent && enqueuedOnPresent) || (enqueuedOnPresent && sequenceNumberPresent)) {
+    throw new TypeError(`${position} is an invalid value for EventPosition. Set only one of offset, sequenceNumber or enqueuedOn properties.`);
+  }
+
+  const validOffset = (typeof position.offset === "number" && position.offset >= -1) || position.offset === latestEventPosition.offset;
+  if (offsetPresent && !validOffset) {
+    throw new TypeError(`${position.offset} is an invalid value for the offset property in EventPosition. Valid values are numbers >= -1 or the string "@latest".`);
+  }
+
+  const validSequenceNumber = typeof position.sequenceNumber === "number" && position.sequenceNumber >= -1;
+  if (sequenceNumberPresent && !validSequenceNumber) {
+    throw new TypeError(`${position.sequenceNumber} is an invalid value for the sequenceNumber property in EventPosition. Valid values are numbers >= -1.`);
+  }
+
+  const validEnqueuedOn = typeof position.enqueuedOn === "number" || position.enqueuedOn instanceof Date;
+  if (enqueuedOnPresent && !validEnqueuedOn) {
+    throw new TypeError(`${position.enqueuedOn} is an invalid value for the enqueuedOn property in EventPosition. Valid values are of type number or instance of Date class.`);
+  }
```

Comment:
@bterlson Please review the above validations after which I'll add the test coverage for the same.

Reply:
https://github.com/Azure/azure-sdk-for-js/pull/6729/commits/c23730b30dc16e1c114f7148f211d6b218d2c306 has the relevant unit tests